### PR TITLE
Run tests from the main controller

### PR DESCRIPTION
**Pull request by kklin**

### CODE REVIEW

File: quilt-tester/tests/spark/check_spark.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+func main() {
+	containers, err := exec.Command("quilt", "containers").Output()
+	if err != nil {
+		log.WithError(err).Fatal("Unable to get containers.")
```
Comment (LuiseV) [inline quilt-tester/tests/spark/check_spark.go]: Is there a reason why we log a Fatal error when `quilt containers` fail, but panic in `check_logs.go` when `quilt machines` fail?

File: quiltctl/ssh/native.go [hunk 2]
```diff
@@ -54,6 +57,40 @@ func (c *NativeClient) Connect(host string, keyPath string) error {
 	return nil
 }
 
+var defaultKeys = []string{"id_rsa", "id_dsa", "id_ecdsa", "id_ed25519"}
+
+// Gets the signers for the default private key locations if possible
+func defaultSigners() []ssh.Signer {
```
Comment (LuiseV) [inline quiltctl/ssh/native.go]: Do we want a unit test for this?

File: quilt-tester/tests/spark/check_spark.go (hunk 1 above)
Comment (kklin) [inline quilt-tester/tests/spark/check_spark.go]: No good reason, I'll change the panic.

File: quiltctl/ssh/native.go (hunk 2 above)
Comment (kklin) [inline quiltctl/ssh/native.go]: I'll add one!

**Comment by kklin** [comment]
@LuiseV Made changes.

Steven's commit is in here because without it, `quilt logs` doesn't work, and thus the spark test wouldn't work.

**Pull request merged**